list_updatable_packages

```diff
@@ -117,15 +130,33 @@ def latest_version(spec: Spec) -> str:
     raise ValueError('Unable to determine latest version', spec)
 
 
-def build_plugin_matrix(specs: Iterable[Spec]) -> Generator[dict, None, None]:
-    if DEBUG: print('Building plugin matrix')
-    for spec in specs:
+def merge_specs(specs: Iterable[Tuple[Spec, Optional[str]]]) -> dict[Spec, Optional[str]]:
+    """
+    Merge specs
+
+    This can process the same spec file multiple times. If a spec is already listed but without a
+    version, it will take the more specific constraint. The last entry is used if a constraint was
+    already given while a warning is printed.
```

Comment:
this ("last entry is used") sounds dangerous as it will potentially update packaging to an incompatible version, while we don't notice the warning?

I'd prefer not touching a package if a conflict was found.

Reply:
I didn't think about that, but skipping it is a good one.

Reply:
I now set the version to `CONFLICT`. Then I expect `./bump_rpm.sh` will fail in the individual job, causing the whole job to fail so we'll find it.

Reply:
smart.

now I wonder if one can upload arbitrary version strings to rubygems, but I think one cannot ;)